### DISCUSSION TITLE
Install new OS to project space

### DISCUSSION THREAD

**Discussion by cleggers21** [procount/pinn#845] [Q&A]
Absolute newbie question …
I’ve at last got PINN running on my Pi5 with 128 gb SD. I want to install 2 versions of the same Pi os & Ubuntu. So I could’nt see a way of saying OSx twice so I created a project space to install the 2nd Pi os later. I’ve both 1 Pi os and Ubuntu on the boot menu but how do I get back to the PINN install screen to install another Pi os onto the project space.
Thanks in anticipation.

**Comment by procount** [comment]
Congrats - yes you are going about it the correct way. Unfortunately, Ubuntu will not allow you to install 2 copies of itself as it uses partition labels as references, and it's not possible to have 2 partitions with the same name 😢. Raspios will be fine, however.

Now that you have installed some (2) OSes, PINN will present the boot menu at startup instead of the PINN install screen. Just before that when the colourful wallpaper is first displayed, you will see a flashing message in the top left corner of the screen. This is when you can press the shift key, click the mouse, joystick button or touch the touchscreen to interrupt the boot process and go back to the install screen.

Replacing a ProjectSpace with a new OS can be a bit confusing, although it makes sense to me.
After installing some OSes, the install screen will automatically check the already installed OSes for selection. In the general case I would suggest clicking the `clear` button to deselect these OSes if appropriate. Then select/check any OSes that you want to install (In your case the Raspios OS), but it could be any other suitable OS. Now go to the Maintenance Menu where you will see the list of installed OSes. Check any OSes that you want to be replaced (e.g. a ProjectSpace). Now you can click the `Replace` button. This will bring up a dialog box listing the existing OSes you want to be replaced. Against each one is a dropdown list of the new OSes you want to install. Just select the appropriate New OS against each existing OS. Once complete, press OK to replace the existing OSes with the new OSes.

Please see https://github.com/procount/pinn/blob/master/README_PINN.md#replace-individual-oses for more details, and the rest of that document for other tips.

**Comment by cleggers21** [comment]
@procount thanks for the very detailed reply. I was pressing <shift> too soon and getting the Rpi boot screen and too late. There is obviously a sweet spot to press <shift> which I was missing but there again, I couldn’t find the boot instructions so wasn’t sure I was doing right thing.
Anyway, all good now, so thanks again for taking the trouble to help me.

**Comment by cleggers21** [comment]
See reply from @procount
